nemo/collections/nlp/modules/common/lm_utils.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2018 The Google AI Language Team Authors and
+# The HuggingFace Inc. team.
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional
+
+from nemo.collections.nlp.modules.common.bert_module import BertModule
+from nemo.collections.nlp.modules.common.huggingface.huggingface_utils import (
+    HUGGINGFACE_MODELS,
+    get_huggingface_lm_model,
+    get_huggingface_pretrained_lm_models_list,
+)
+from nemo.collections.nlp.modules.common.megatron.megatron_utils import (
+    get_megatron_lm_model,
+    get_megatron_lm_models_list,
+)
+from nemo.utils import logging
+
+__all__ = ['get_pretrained_lm_models_list', 'get_lm_model']
+
+
+def get_pretrained_lm_models_list() -> List[str]:
+    """
+    Returns the list of supported pretrained model names
+    """
+    return get_megatron_lm_models_list() + get_huggingface_pretrained_lm_models_list()
+
+
+def get_lm_models_list() -> List[str]:
+    """
+    Returns the list of supported models based on the model type
+    """
+    return ["megatron"] + list(HUGGINGFACE_MODELS.keys())
```

Comment:
can we have the complete list of megatron names back here?

Reply:
since model_type will be removed, this function will be gone too

Reply:
wait, why is this function will be gone? we need a way to see the list of available models

Reply:
there is another function get_pretrained_lm_models_list